lib/ruby_smb/client/negotiation.rb

```diff
@@ -66,6 +65,11 @@ def negotiate_response(raw_data)
         response = nil
         if smb1
           packet = RubySMB::SMB1::Packet::NegotiateResponseExtended.read raw_data
+
+          unless packet.valid?
+            packet = RubySMB::SMB1::Packet::NegotiateResponse.read raw_data
+          end
+
```

Comment:
what's this doing? if the packet isn't valid you read it again? seems a bit weird is this common?

Reply:
No worries, I made that mistake a bunch when I started on this 😄